apps/web/src/app/[locale]/challenge/[id]/comments/[commentId]/page.tsx

```diff
@@ -23,6 +23,7 @@ export default async function CommentPage({ params: { id, commentId } }: Props)
     <div className="relative h-full">
       <Description challenge={challenge} />
       <Comments
+        challengeId={Number(id)}
```

Comment:
isnt this just the same value as `rootId` ?

Reply:
Correct, in this case,
but not for the `Solution Comment` path.

I felt the `rootId` introduced some confusion for me later in the code.
So making this data duplicate but explicit was a trade off I made.

Reply:
AHA! good enough for now tbh

Reply:
actually can’t you just grab challenge id from the url via useParams. that way you don’t need to save the challengeId or anything i think

Reply:
> actually can’t you just grab challenge id from the url via useParams. that way you don’t need to save the challengeId or anything i think
> 
> would also avoid having only new comments be linkable which is already sus imo

You're right, its cleaner to grab it from the params in `<Comments />`

I was hesitant to do this before, Because `<Comments />` lives in `_components/` and not `app/` typescript doesn't know what `params` will actually exist.

Will make the necessary changes and test, thanks!